<html>
<head>
<title>Ontology Alignment Evaluation Initiative::Anatomy results</title>
<link rel="stylesheet" type="text/css" href="../../style.css" />
</head>
<body>

<div class="header">
<a style="color: grey; line-height: 5mm;" href="https://oaei.ontologymatching.org/2010/">Ontology
  Alignment Evaluation Initiative - OAEI-2010
  Campaign</a><a href="https://oaei.ontologymatching.org/"><img src="../../../oaeismall.jpg" alt="OAEI"
		   style="float:right; margin-left: 5pt; border-style:none;"/></a>
</div>

<h1>Anatomy - Results of 2010 Evaluation</h1>

<p>
The anatomy track confronts existing matching technology with a specific type of real world ontologies from the biomedical domain. In this domain a significant number of ontologies have been built covering different aspects of medical research.
</p>

<h2>Test Data and Experimental Setting</h2>

<p>
The data set of this track has been used since 2007. For a detailed description we refer the reader to the OAEI 2007 [2] results paper. The ontologies of the anatomy track are the NCI Thesaurus describing the human anatomy, published by the National Cancer Institute (NCI), and the Adult Mouse Anatomical Dictionary, which has been developed as part of the Mouse Gene Expression Database project. Both resources are part of the Open Biomedical Ontologies (OBO). The alignment between these ontologies has been created by experts of the domain [1].
</p>

<p>
As in the previous years, we divided the matching task into four subtasks. Subtask #1 is obligatory for participants of the anatomy track, while subtask #2, #3 and #4 are again optional tasks.
</p>

<ul>
	<li>#1 The matcher has to be applied with its standard settings.</li>
	<li>#2 An alignment has to be generated that favors precision over recall.</li>
	<li>#3 An alignment has to be generated that favors recall over precision.</li>
	<li>#4 A partial reference alignment has to be used as additional input.</li>	
</ul>


<p>
Notice that in 2010 we used the SEALS evaluation service for subtask #1. In the course of using the SEALS services we published the complete reference alignment for the first time. In the future we plan to include all subtasks in the SEALS modality. This requires to extend the interfaces of the SEALS evaluation service to allow for example an (incomplete) alignment as additional input parameter.
</p>

<p>
The harmonization of the ontologies applied in the process of generating a reference alignment (see [1] and [2 resulted in a high number of rather trivial correspondences (61%). These correspondences can be found by very simple string comparison techniques. At the same time, we have a good share of non-trivial correspondences (39%). This is an important characteristic of the data set to be taken into account in the following analysis. The partial reference alignment used in subtask #4 is the union of all trivial correspondences and 54 non-trivial correspondences.
</p>

<p>
Due the experiences made in the past, we decided to slightly modify the test data set for the 2010 evaluation. We removed some doubtful subsumption correspondences and added a number of disjointness statement at the top of the hierarchies to increase the expressivity of the data set. Furthermore, we  eliminated three incorrect correspondences. The reference alignment is now coherent with respect to the ontologies to be matched<!-- FOOTNOTE-->.
</p>

<h2>Results</h2>

<p>
While the number of participants is nearly stable over four years, we find in 2010 more systems that participated for the first time (5 systems) than in the previous years (in average 2 systems). An overview is presented in the table below. Four of the newcomers participate also in other tracks, while NBJLM participates only in the Anatomy track. NBJLM is thus together with AgreementMaker (AgrMaker) a system that uses a track-specific parameter setting. Taking part in several tracks with a standard setting makes it obviously much harder to obtain good results in a specific track.
</p>

<table border="1" cellpadding="6" cellspacing="1">
  <tr>
    <th> System  </th><th> 2007  </th><th> 2008  </th><th> 2009 </th><th> 2010
</th></tr>
<tr><td>
AFlood   </td><td>   &nbsp;    </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td> &nbsp;   
</td></tr><tr><td>
AgrMaker </td><td> <img src="tick.gif"></img> </td><td>  &nbsp;      </td><td> <img src="star.gif"></img> </td><td> <img src="star.gif"></img>
</td></tr><tr><td>
AROMA     </td><td>    &nbsp;      </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td>  &nbsp;   
</td></tr><tr><td>
AOAS     </td><td> <img src="star.gif"></img></td><td>   &nbsp;        </td><td>   &nbsp;       </td><td> &nbsp;   
</td></tr><tr><td>
ASMOV   </td><td> <img src="tick.gif"></img> </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td> <img src="tick.gif"></img> 
</td></tr><tr><td>
BLOOMS    </td><td>   &nbsp;       </td><td>  &nbsp;         </td><td> &nbsp;     </td><td> <img src="star.gif"></img> 
</td></tr><tr><td>
CODI    </td><td>    &nbsp;      </td><td>    &nbsp;       </td><td>   &nbsp;   </td><td> <img src="tick.gif"></img> 
</td></tr><tr><td>
DSSim   </td><td> <img src="tick.gif"></img> </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td>  &nbsp; 
</td></tr><tr><td>
Ef2Match   </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>   &nbsp;   </td><td> <img src="star.gif"></img> 
</td></tr><tr><td>
Falcon AO  </td><td> <img src="tick.gif"></img> </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>  &nbsp; 
</td></tr><tr><td>
GeRMeSMB    </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>   &nbsp;   </td><td> <img src="tick.gif"></img> 
</td></tr><tr><td>
Kosimap    </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td> <img src="tick.gif"></img> </td><td>  &nbsp; 
</td></tr><tr><td>
Lily     </td><td> <img src="tick.gif"></img> </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td>  &nbsp; 
</td></tr><tr><td>
NBJLM   </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>  &nbsp;    </td><td> <img src="star.gif"></img> 
</td></tr><tr><td>
Prior+  </td><td> <img src="tick.gif"></img> </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>  &nbsp; 
</td></tr><tr><td>
RiMOM   </td><td> <img src="tick.gif"></img> </td><td> <img src="star.gif"></img> </td><td> <img src="tick.gif"></img> </td><td>  &nbsp; 
</td></tr><tr><td>
SAMBO  </td><td> <img src="star.gif"></img> </td><td> <img src="star.gif"></img> </td><td>  &nbsp;      </td><td>  &nbsp; 
</td></tr><tr><td>
SOBOM   </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td> <img src="star.gif"></img> </td><td> <img src="star.gif"></img> 
</td></tr><tr><td>
TaxoMap  </td><td> <img src="tick.gif"></img> </td><td> <img src="tick.gif"></img>  </td><td> <img src="tick.gif"></img> </td><td> <img src="star.gif"></img> 
</td></tr><tr><td>
X SOM   </td><td> <img src="tick.gif"></img> </td><td>  &nbsp;      </td><td>  &nbsp;      </td><td>   &nbsp; 
</td></tr><tr><td>
Avg. f-measure  </td><td> 0.598 </td><td> 0.718  </td><td> 0.764  </td><td> 0.785 
</td></tr>
<caption align="bottom"><i>Particpants of the last years.</i><caption>
</table>

<p>
In the last row of the table the average of f-measures per year in subtask #1 is shown. We observe significant improvements over time. However, the measured improvements decrease over time and seem to reach a top (2007 +</img>12% -&gt; 2008 <+5% -&gt; 2009 +2% -&gt; 2010). We have marked the participants with an f-measure &gt= 0.8 with a <img src="star.gif"></img> symbol.
</p>


<h3>Runtime</h3>

<p>
In the previous years we reported about runtimes that have been measured by the participants. The differences we observed  - from several minutes to several days - could not be explained by the use of different hardware. However, these differences became less significant over the years and in 2009 all systems except one required between 2 and 30 minutes. Therefore, we abstained from an analysis of runtimes this year. In 2011 we plan to execute the matching systems on the SEALS platform to enable an exact measurement of runtimes not biased by differences in hardware equipment. So far we refer the reader interested in runtimes to the result papers of the participants. Results papers can be found in the <a href="http://CEUR-WS.org/Vol-689/">Ontology matching workshop proceedings</a>.
</p>

<h3>Main results for subtask #1</h3>

<p>
The results for subtask #1 are presented in the following tableordered with respect to the achieved f-measure. In 2010 AgreementMaker (AgrMaker) generates the best alignment with respect to f-measure. Moreover, this result is based on a high recall compared to the systems on the following positions. This is a remarkable result, because even the SAMBO system of 2008 could not generate a higher recall with the use of UMLS. However, we have to mention again that AgreementMaker uses a specific setting for the anatomy track.
</p>


<table border="1" cellpadding="6" cellspacing="1">
  <tr>
    <th colspan="1">System</th>
    <th colspan="3">Task #1</th>
    <th colspan="3">Task #2</th>
    <th colspan="3">Task #3</th>
    <th colspan="2">Recall+</th>
  </tr>
<tr><td> System</td><td> Precision </td><td> Recall </td><td> F-measure </td><td> Precision </td><td> Recall </td><td> F-measure </td><td> Precision </td><td> Recall </td><td> F-measure </td><td> Subtask #1 </td><td> Subtask #3 </td></tr><tr><td> 
AgrMaker* </td><td> 0.903 </td><td> 0.853 </td><td> 0.877 </td><td> 0.962 </td><td> 0.751 </td><td> 0.843 </td><td> 0.771 </td><td> 0.874 </td><td> 0.819 </td><td> 0.630 </td><td> 0.700  
</td></tr>


<tr>
<td> 
Ef2Match </td><td> 0.955 </td><td> 0.781 </td><td> 0.859 </td><td>  0.968  </td><td>  0.745  </td><td>  0.842  </td><td>  0.954 (0.766) </td><td>  0.781 (0.838) </td><td> 0.859 (0.800 ) </td><td> 0.440 </td><td>  0.440 (0.588)   
</td></tr>
<tr><td>NBJLM* </td><td> 0.920 </td><td> 0.803 </td><td> 0.858 </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td> 0.569 </td><td>  -   
</td></tr><tr><td> 
SOBOM </td><td> 0.949 </td><td> 0.778 </td><td> 0.855 </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td>  -  </td><td> 0.433 </td><td>  -   
</td></tr><tr><td> 
BLOOMS </td><td> 0.954 </td><td> 0.731 </td><td> 0.828 </td><td> 0.967 </td><td> 0.725 </td><td> 0.829 </td><td>  -  </td><td>  -  </td><td>  -  </td><td> 0.315 </td><td>  -   
</td></tr><tr><td> 
TaxoMap </td><td> 0.924 </td><td> 0.743 </td><td> 0.824 </td><td> 0.956 </td><td> 0.689 </td><td> 0.801 </td><td> 0.833 </td><td> 0.774 </td><td> 0.802 </td><td> 0.336 </td><td> 0.414  
</td></tr><tr><td> 
ASMOV </td><td> 0.799 </td><td> 0.772 </td><td> 0.785 </td><td> 0.865 </td><td> 0.757 </td><td> 0.808 </td><td> 0.717 </td><td> 0.792 </td><td> 0.753 </td><td> 0.470 </td><td> 0.538  
</td></tr><tr><td> 
CODI </td><td> 0.968 </td><td> 0.651 </td><td> 0.779 </td><td> 0.964 </td><td> 0.662 </td><td> 0.785 </td><td> 0.782 </td><td> 0.695 </td><td> 0.736 </td><td> 0.182 </td><td> 0.383  
</td></tr><tr><td>
GeRMeSMB </td><td> 0.884 </td><td> 0.307 </td><td> 0.456 </td><td> 0.883 </td><td> 0.307 </td><td> 0.456 </td><td> 0.080 </td><td> 0.891 </td><td> 0.147 </td><td> 0.249 </td><td> 0.838  </td></tr>
<caption align="bottom"><i>Results for subtask #1, #2, and #3 in terms of precision, recall (in addition recall+ for #1 and #3) and f-measure. Systems marked with a * do not participate in other tracks or have chosen a setting specific to this track. Note that ASMOV modified its standard settting in a restricted way (activating UMLS as additional resource). Thus, we did not mark this system. Some values for Ef2Match have been added after the final submission deadline <a href="#corrections">(see below)</a></i>
</caption>

</table>

<p>
AgreementMaker is followed by three participants (Ef2Match, NBJLM, SOBOM) that share a very similar characteristic regarding f-measure and observed precision score. All of these systems clearly favor precision over recall. A further analysis has to clarify to which degree the alignments generated by these systems are overlapping as indicated by their precision/recall characteristics. Notice that these systems obtained better scores or scores that are similar to the results of the top systems in the previous years. One explanation can be seen in the fact that the organizers of the track made the reference alignment available to the participants. More precisely, participants could at any time compute precision and recall scores via the SEALS services to test different settings of their algorithms. On the one hand this allows to improve a matching system by a constant formative evaluation in a direct feedback cycle, on the other hand it might happen that a perfect configuration results in problems for different data sets. 
</p>


 
<h3>Recall+ and further results</h3>

<p>
In the following we use again the recall+ measure as defined in [2]. It measures how many non trivial correct correspondences, not detectable by string equivalence, can be found in an alignment. The top three systems with respect to recall+ regarding subtask #1 are AgreementMaker, NBJLM and ASMOV. Only ASMOV has participated in several tracks with the same setting. Obviously, it is not easy to find a large amount of non-trivial correspondences with a standard setting.
</p>

<p>
In 2010 six system participated in subtask #3. The top three systems regarding recall+ in this task are GeRoMe-SMB (GeRMeSMB), AgreementMaker and ASMOV. Since a specific instruction about the balance between precision and recall is missing in the description of the task, the results vary to a large degree. GeRoMe-SMB detected 83.8% of the correspondences marked as non-trival, but at a precision of 8\%. AgreementMaker and ASMOV modified their settings only slightly, however, they were still able to detect 70% and 53.8% of all non trivial correspondences.
</p>

<p>
In subtask #2 seven systems participated. It is interesting to see that systems like ASMOV, BLOOMS and CODI generate alignments with slightly higher f-measure for this task compared to the submission for subtask #1. The results for subtask #2 for AgreementMaker are similar to the results submitted by other participants for subtask #1. This shows that many systems in 2010 focused on a similar strategy that exploits the specifics of the data set resulting in a high f-measure.
</p>

<p>
Only about half of the participants submitted results for subtask #2 and #3. This can be related to an unclear description of the expected results. In the future we have to think about an alternative description of the subtask together with a different kind of evaluation to increase participation.
</p>

 
<h3>Results for subtrack #4</h3>

<p>

In the following we refer to an alignment generated for subtask #n as  A<sub>n</sub>. In our evaluation we use again the method introduced in 2009. We compare both A<sub>1</sub> &cup; R<sub>p</sub> and A<sub>4</sub> &cup; R<sub>p</sub> with the reference alignment R. Thus, we compare the situation where the partial reference alignment is added after the matching process against the situation where the partial reference alignment is available as additional resource exploited within the matching process. Note that a direct comparison of A<sub>1</sub> and A<sub>4</sub> would not take into account in how far the partial reference alignment was already included in A<sub>1</sub> resulting in a distorted interpretation.
</p>




<p>
Results are presented in the following table. Three systems participated in task #4 in 2010. Additionally, we added a row for the 2008 submission of SAMBOdtf. This system had the best results measured in the last years. AgreementMaker and ASMOV use the input alignment to increase the precision of the final result. At the same time these systems filter out some correct correspondences, finally resulting in a slightly increased f-measure. This fits with the tendency we observed in the past years (compare with the results for SAMBOdtf in 2008). The effects of this strategy are not very strong. However, as argued in the previous years, the input alignment has a characterictis that makes it hard to exploit this information.
</p>


<table border="1" cellpadding="6" cellspacing="1">
<tr>
	<th>System</th>
	<th>&Delta;-Precision</th>
	<th>&Delta;-Recall</th>
	<th>&Delta;-Fmeasure</th>
</tr>
<tr>
<td> AgrMaker </td><td> 0.025  <span style="font-size:small">(0.904 &rarr; 0.929)</span> </td><td> 0.025  <span style="font-size:small">(0.876 &rarr; 0.851)</span> </td><td> 0.002 <span style="font-size:small">(0.890 &rarr; 0.888)</span> </td>
</tr>

<tr>
<td> ASMOV </td><td> 0.029  <span style="font-size:small">(0.808 &rarr; 0.837)</span> </td><td> 0.016 <span style="font-size:small">(0.824 &rarr; 0.808)</span> </td><td> 0.006 <span style="font-size:small">(0.816 &rarr; 0.822)</span> </td>
</tr>

<tr>
<td> CODI </td><td> 0.002 <span style="font-size:small">(0.970 &rarr; 0.968)</span> </td><td> 0.030 <span style="font-size:small">(0.716 &rarr; 0.746)</span> </td><td> 0.019 <span style="font-size:small">(0.824 &rarr; 0.843)</span> </td>
</tr>

<tr>
<td> SAMBOdtf in 2008</span> </td><td> 0.021 <span style="font-size:small">(0.837 &rarr; 0.856)</span> </td><td> 0.003 <span style="font-size:small">(0.867 &rarr; 0.870)</span> </td><td> 0.011 <span style="font-size:small">(0.852 &rarr; 0.863)</span> </td>
</tr>
<caption align="bottom"><i>Changes in precision, recall and f-measure based on comparing subtask #1 and #4.</i>
</caption>
</table>

<p>
CODI has chosen a different strategy. While changes in precision are negligible, recall increases by 3%. Even though the overall effect is still not very strong, the system exploits the input alignment in the most effective way. However, the recall of CODI for subtask #1 is relatively low compared to the other systems. It is unclear whether the strategy of CODI would also work for the other systems where a ceiling effect might prevent the exploitation of the positive effects. We refer the interested reader to the results paper of the system for a description of the algorithm.
</p>


<h2>Conclusions</h2>

<p>
Overall, we see a clear improvement comparing this years results with the results of the previous years. This holds both for the average participant as well as for the top performer. A very positive outcome can be seen in the increased recall values. In addition to the evaluation experiments we reported, we computed the union of all submissions to subtask #1. For the resulting alignment we measured a precision of 69.7% and a recall of 92.7%. We added additionaly the correct correspondences generated in subtask #3 and reached a recall of 97.1%. Combining the strategies used by different matching systems it is thus possible to detect nearly all correct correspondences.
</p>

<p>
The availability of the SEALS evaluation service surely had an effect on the results submitted in 2010. We have already argued about pros and cons. In the future we plan to extend the data set of the anatomy track with additional ontologies and reference alignments to a more comprehensive  and general track covering different types of biomedical ontologies. In particular, we will not publish the complete set of reference alignments to conduct a part of the evaluation experiment in the blind mode. This requires, however, to find and analyze interesting and well-suited data sets. The strategy to publish parts of the evaluation material and to keep other parts hidden seems to be the best approach.
</p>

<p>&nbsp;</p>

<hr/>

<p><i><b> The paragraphs above the horizontal rule is to 99% the content presented in the OAEI preliminary results paper. If you detect any errors or misleading interpretations, please report them to Christian Meilicke (address below). It is then possible to fix these issues directly in this webpage and then also in the final version of the paper.</b></i></p>

<h2>2009 vs. 2010 version</h2>

<p>We have above reported about slight differences in 2010 and 2009 version. To check in how far these changes might have affected the results, we applied the matcher AROMA and AFlood in 2009 setting on both version of the datasets and AgreementMaker (thanks to Cosmin Stroe!) with 2010 setting on both versions of the dataset. Results are presented in the following table.</p>


<table border="1" cellpadding="6" cellspacing="1">
<tr>
	<th>Matcher/Setting </th>
	<th>precision 09 &rarr; 10 </th>
	<th>recall 09 &rarr; 10 </th>
	<th>f-measure 09 &rarr; 10 </th>
	<th>|A<sub>09</sub>-A<sub>10</sub>| + |A<sub>10</sub>-A<sub>09</sub>|</th>
</tr>

<tr>
<td>Aroma (2009 setting)</td><td>0.770 &rarr; 0.769</td><td>0.678 &rarr; 0.683</td><td>0.721 &rarr; 0.724</td><td>33</td>
</tr>

<tr>
<td>AFlood (2009 setting)</td><td>0.873 &rarr; 0.871</td><td>0.653 &rarr; 0.664</td><td>0.747 &rarr; 0.753</td><td>173</td>
</tr>

<tr>
<td>AgrMaker (2010 setting)</td><td>0.896 &rarr; 0.903</td><td>0.848 &rarr; 0.853</td><td>0.872 &rarr; 0.877</td><td>128</td>
</tr>
</table>

<p>Note that the differences in terms of precision, recall and f-measure are very small. It seems that the fixes helped matching systems to a very limited degree (even though the generated alignments themselves show some more significant differences in number of correspondences that do not overlap). However, AgreementMaker would also for the 2009 version of the dataset generate the alignment with the highest f-measure and very similar precision/recall characteristic.</p>

<h2>Final Submissions of the 2010 participants</h2>

<p><a href="oaei2010-anatomysubmissions.zip">Download alignments</a></p>

<h2><a name="corrections" id="corrections">Modifications / Updates / Corrections</a></h2>

<p><b>(1)</b>
Chua Wei Khong Watson reported that his submission of Eff2Match to subtasks #2 and #3 have not been included in the results. This was a mistake of the organizers and will be fixed in the final version of the OAEI results paper. We already applied the corresponding changes to this webpage.
</p>

<p><b>(2)</b>
Chua Wei Khong Watson reported that his submission of Eff2Match to subtasks #3 was unfortunately based on a wrong file. The correct file has now been analysed and the corrected values have been added in parentheses.
</p>


<h2>Contact</h2>

<p>This track is organized by <a href="http://ki.informatik.uni-mannheim.de/people/christian_meilicke.html">Christian Meilicke</a> and <a href="http://ki.informatik.uni-mannheim.de/people/heiner_stuckenschmidt.html">Heiner Stuckenschmidt</a>.
If you notice any errors or misleading  remarks on this page, directly contact Christian Meilicke (email to christian [at] informatik [.] uni-mannheim [.] de).</p>

<h3>Acknowledgements</h3>

<p><i>
We gratefully thank Elena Beisswanger (Jena University Language and Information Engineering Lab) for her thorough support on improving the quality of the data set. The modifications are documented <a href="../../anatomy/modifications2010.html">here</a>.
</i>
</p>


<h3><a name="references" id="references">References</a></h3>

<p>
[1] Bodenreider O. and Hayamizu T. and Ringwald M. and De Coronado S. and Zhang S.: Of Mice and Men: Aligning Mouse and Human Anatomies. Proceedings of the American Medical Informatics Association (AIMA) Annual Symposium, 2005.
</p>


<p>
[2] Jérôme Euzenat et. al.: Results of the ontology alignment evaluation initiative 2007. In Proceedings 2nd International Workshop on Ontology Matching (OM-07), Busan (Korea), 2007.
</p>



<div class="address">
Original page: <a href="http://webrum.uni-mannheim.de/math/lski/anatomy10/results.html">http://webrum.uni-mannheim.de/math/lski/anatomy10/results.html</a> [cached: 08/03/2011]
<div class="footer">https://oaei.ontologymatching.org/2010/results/anatomy/</div>
<!--Id-->
</div>
</body>
</html>
